google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/converter/AppEngineFacetMigrationConversionProcessor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.converter;
+
+import com.google.cloud.tools.intellij.appengine.facet.AppEngineFacetType;
+
+import com.intellij.conversion.CannotConvertException;
+import com.intellij.conversion.ConversionProcessor;
+import com.intellij.conversion.ModuleSettings;
+
+import org.jdom.Element;
+import org.jetbrains.jps.model.serialization.facet.JpsFacetSerializer;
+
+/**
+ * Performs conversions from deprecated App Engine Facets to their new version.
+ */
+public class AppEngineFacetMigrationConversionProcessor extends
+    ConversionProcessor<ModuleSettings> {
+
+  static final String DEPRECATED_APP_ENGINE_FACET_ID = "google-app-engine";
```

Comment:
Personally I would avoid aggressive factoring of test code and optimize for clarity of what you're testing and expecting. Deep calls stacks in the actual test code should be avoided. You could just duplicate the facet ID into a constant on the test class.  

For cases where you don't want to do that (and I've probably reached into a constant like this one before for testing purposes), just mark the field @VisibleForTesting to indicate why the visibility is relaxed.